samples: flash_shell: Enable on all platforms with a flash driver

galak opened the pull request:
Try and build the flash_shell on all platforms that have a flash driver
rather than a limited set of know platforms.  This hopefully acts as a
build coverage test for all the flash drivers.

The flash shell requires around 10K of memory so limit it to systems
with 12K or more.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
samples: flash_shell: Enable on all platforms with a flash driver

Try and build the flash_shell on all platforms that have a flash driver
rather than a limited set of know platforms.  This hopefully acts as a
build coverage test for all the flash drivers.

The flash shell requires around 10K of memory so limit it to systems
with 12K or more.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
@MaureenHelm going to open an issue related to this, but this test shows a build issue with flash and KW4x.

On hexiwear_kw40z:

```
make[3]: *** [zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/build.make:63: zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/s
make[3]: Target 'zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/build' not remade because of errors.
make[3]: Leaving directory '/home/galak/git/zephyr/sanity-out/hexiwear_kw40z/samples/drivers/flash_shell/test'
make[2]: *** [CMakeFiles/Makefile2:2254: zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/all] Error 2
[ 91%] Linking C static library app/libapp.a
make[3]: Leaving directory '/home/galak/git/zephyr/sanity-out/hexiwear_kw40z/samples/drivers/flash_shell/test'
[ 91%] Built target app
[ 92%] Linking C static library libkernel.a
make[3]: Leaving directory '/home/galak/git/zephyr/sanity-out/hexiwear_kw40z/samples/drivers/flash_shell/test'
[ 92%] Built target kernel
In file included from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_features.h:21:0,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_controller.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_flash.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_flash.c:10:
/home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_adapter.h:302:0: error: "SIM_FCFG1_REG" redefined [-Werror]
 #define SIM_FCFG1_REG SIM->FCFG1
 
In file included from /home/galak/git/zephyr/ext/hal/nxp/mcux/devices/MKW40Z4/fsl_device_registers.h:44:0,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_common.h:22,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_features.h:18,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_controller.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_flash.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_flash.c:10:
/home/galak/git/zephyr/ext/hal/nxp/mcux/devices/MKW40Z4/MKW40Z4.h:6829:0: note: this is the location of the previous definition
 #define SIM_FCFG1_REG(base)                      ((base)->FCFG1)
 
In file included from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_features.h:21:0,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_controller.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_controller.c:10:
/home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_adapter.h:302:0: error: "SIM_FCFG1_REG" redefined [-Werror]
 #define SIM_FCFG1_REG SIM->FCFG1
 
In file included from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_features.h:21:0,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_controller.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_flash.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_flash.c:10:
/home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_flash.c: In function ‘FLASH_Init’:
/home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_adapter.h:376:55: error: ‘FSL_FEATURE_FLASH_PFLASH_PROTECTION_REGIO
 #define FLASH0_FEATURE_PFLASH_PROTECTION_REGION_COUNT FSL_FEATURE_FLASH_PFLASH_PROTECTION_REGION_COUNT
                                                       ^
/home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_adapter.h:376:55: note: in definition of macro ‘FLASH0_FEATURE_PFLA
 #define FLASH0_FEATURE_PFLASH_PROTECTION_REGION_COUNT FSL_FEATURE_FLASH_PFLASH_PROTECTION_REGION_COUNT
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_adapter.h:376:55: note: each undeclared identifier is reported only
 #define FLASH0_FEATURE_PFLASH_PROTECTION_REGION_COUNT FSL_FEATURE_FLASH_PFLASH_PROTECTION_REGION_COUNT
                                                       ^
/home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_adapter.h:376:55: note: in definition of macro ‘FLASH0_FEATURE_PFLA
 #define FLASH0_FEATURE_PFLASH_PROTECTION_REGION_COUNT FSL_FEATURE_FLASH_PFLASH_PROTECTION_REGION_COUNT
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/galak/git/zephyr/ext/hal/nxp/mcux/devices/MKW40Z4/fsl_device_registers.h:44:0,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_common.h:22,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_features.h:18,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_controller.h:13,
                 from /home/galak/git/zephyr/ext/hal/nxp/mcux/drivers/kinetis/fsl_ftfx_controller.c:10:
/home/galak/git/zephyr/ext/hal/nxp/mcux/devices/MKW40Z4/MKW40Z4.h:6829:0: note: this is the location of the previous definition
 #define SIM_FCFG1_REG(base)                      ((base)->FCFG1)
 
```

erwango commented on samples/drivers/flash_shell/sample.yaml:
```diff
@@ -4,6 +4,7 @@ sample:
   test:
-    platform_whitelist: 96b_carbon frdm_k64f frdm_kw41z frdm_kl25z nucleo_f746zg
     tags: flash shell
+    filter: CONFIG_FLASH_HAS_DRIVER_ENABLED
```
Does this mean that before running sanitycheck, we're building cmake config for all available boards?

MaureenHelm replied:
Why are we handling this with a Kconfig filter rather than `depends_on`?

nashif replied:
good point, I agree we should use depends_on here and avoid filters on Kconfig

galak replied:
I purposefully didn't want to add "flash" to all the board yaml.  The CONFIG_ option makes more sense to me.

MaureenHelm commented on samples/drivers/flash_shell/prj.conf:
```diff
@@ -7,4 +7,4 @@ CONFIG_FLASH=y
 # it here.
 # CONFIG_FLASH_PAGE_LAYOUT=y
-CONFIG_MPU_ALLOW_FLASH_WRITE=y
+#CONFIG_MPU_ALLOW_FLASH_WRITE=y
```
Just remove the line instead of commenting it out.

galak committed:
flash: nxp_lpc: Fix build issue with LPC Flash driver

The flash driver wasn't getting built in the NXP HAL.  Pull in version
of HAL that fixes that.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
flash: sam0: Fix build issue.

Fix stale DT define that causes driver not to build.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
flash: sam: Limit flash driver to SAME70 SoCs

The SAM flash driver was developed for the SAME70 SoCs and hasn't ever
been validated on the other SAM families (SAM3X, SAM4S).  If we try and
build on one of the other SoC families we get build errors.  So limit it
to just SAME70 for now.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
@nashif I'd ask we rethink the comment about depends_on vs filter.  As we want to move away from the board yaml and use DT, this would end up being:

```
filter: 

dt_compat_enabled("atmel,sam-flash-controller") or 
dt_compat_enabled("atmel,sam0-nvmctrl") or 
dt_compat_enabled("cypress,psoc6-flash-controller") or 
dt_compat_enabled("nordic,nrf-qspi") or 
dt_compat_enabled("nordic,nrf51-flash-controller") or 
dt_compat_enabled("nordic,nrf52-flash-controller") or 
dt_compat_enabled("nordic,nrf53-flash-controller") or 
dt_compat_enabled("nordic,nrf91-flash-controller") or 
dt_compat_enabled("nxp,kinetis-ftfa") or 
dt_compat_enabled("nxp,kinetis-ftfe") or 
dt_compat_enabled("nxp,kinetis-ftfl") or 
dt_compat_enabled("nxp,lpc-iap") or 
dt_compat_enabled("openisa,rv32m1-ftfe") or 
dt_compat_enabled("silabs,gecko-flash-controller") or 
dt_compat_enabled("st,stm32f0-flash-controller") or 
dt_compat_enabled("st,stm32f1-flash-controller") or 
dt_compat_enabled("st,stm32f2-flash-controller") or 
dt_compat_enabled("st,stm32f3-flash-controller") or 
dt_compat_enabled("st,stm32f4-flash-controller") or 
dt_compat_enabled("st,stm32f7-flash-controller") or 
dt_compat_enabled("st,stm32g0-flash-controller") or 
dt_compat_enabled("st,stm32g4-flash-controller") or 
dt_compat_enabled("st,stm32h7-flash-controller") or 
dt_compat_enabled("st,stm32l1-flash-controller") or 
dt_compat_enabled("st,stm32l4-flash-controller") or 
dt_compat_enabled("st,stm32wb-flash-controller")
```

nashif commented:
> @nashif I'd ask we rethink the comment about depends_on vs filter. As we want to move away from the board yaml and use DT, this would end up being:
> 
> ```
> filter: 
> 
> dt_compat_enabled("atmel,sam-flash-controller") or 
> dt_compat_enabled("atmel,sam0-nvmctrl") or 
> dt_compat_enabled("cypress,psoc6-flash-controller") or 
> dt_compat_enabled("nordic,nrf-qspi") or 
> dt_compat_enabled("nordic,nrf51-flash-controller") or 
> dt_compat_enabled("nordic,nrf52-flash-controller") or 
> dt_compat_enabled("nordic,nrf53-flash-controller") or 
> dt_compat_enabled("nordic,nrf91-flash-controller") or 
> dt_compat_enabled("nxp,kinetis-ftfa") or 
> dt_compat_enabled("nxp,kinetis-ftfe") or 
> dt_compat_enabled("nxp,kinetis-ftfl") or 
> dt_compat_enabled("nxp,lpc-iap") or 
> dt_compat_enabled("openisa,rv32m1-ftfe") or 
> dt_compat_enabled("silabs,gecko-flash-controller") or 
> dt_compat_enabled("st,stm32f0-flash-controller") or 
> dt_compat_enabled("st,stm32f1-flash-controller") or 
> dt_compat_enabled("st,stm32f2-flash-controller") or 
> dt_compat_enabled("st,stm32f3-flash-controller") or 
> dt_compat_enabled("st,stm32f4-flash-controller") or 
> dt_compat_enabled("st,stm32f7-flash-controller") or 
> dt_compat_enabled("st,stm32g0-flash-controller") or 
> dt_compat_enabled("st,stm32g4-flash-controller") or 
> dt_compat_enabled("st,stm32h7-flash-controller") or 
> dt_compat_enabled("st,stm32l1-flash-controller") or 
> dt_compat_enabled("st,stm32l4-flash-controller") or 
> dt_compat_enabled("st,stm32wb-flash-controller")
> ```

what is the above exactly? Is this how it would end up working with DT? I hope I am reading this wrong :)

nashif commented:
oops, this had a west.yaml change I have just noticed, should have been marked with DNM :(

erwango commented on west.yml:
```diff
@@ -86,7 +86,7 @@ manifest:
       path: tools/net-tools
     - name: hal_nxp
-      revision: 62b46313b8e07e2d5062fd62f14ee9dfeb201b76
+      revision: pull/29/head
```
@galak 
This should not have been merged as is

nvlsianpu replied:
would be nice to have check against pull reference committing in the CI.

erwango replied:
Has been fixed afterwards it seems.
